src/videojs-hls.js

```diff
@@ -486,31 +487,32 @@ videojs.Hls.prototype.seekable = function() {
 /**
  * Update the player duration
  */
-videojs.Hls.prototype.updateDuration = function(playlist) {
+videojs.HlsHandler.prototype.updateDuration = function(playlist) {
   var oldDuration = this.mediaSource.duration,
       newDuration = videojs.Hls.Playlist.duration(playlist),
       setDuration = function() {
         this.mediaSource.duration = newDuration;
         this.tech_.trigger('durationchange');
+
+        // update seekable
+        if (seekable.length !== 0 && newDuration === Infinity) {
```

Comment:
Below we trigger `durationchange` after setting seekable; here we do it before. Since there is no `seekablechange` event, we probably want to trigger `durationchange` afterwards here as well for now.